### PR TITLE
Fix error with Python3.7 now PEP479 is enforced.

### LINKED ISSUE #130: Enforcing PEP479 Cause RuntimeError:  generator raised StopIteration in _gen_georss_coords

### PR DESCRIPTION
**Pull request by rgammans**
- Change _gen_georss_coords to capture StopIteration
   from value list, using pattern from PEP

Fixes #130

### CODE REVIEW

File: feedparser/namespaces/georss.py [hunk 1]
```diff
@@ -175,12 +175,15 @@ def _parse_poslist(value, geom_type, swap=True, dims=2):
...
-        if dims == 3:
-            t.append(next(latlons))
-        yield tuple(t)
+        try:
+            t = [next(latlons), next(latlons)][::swap and -1 or 1]
+            if dims == 3:
+                t.append(next(latlons))
+            yield tuple(t)
+        except StopIteration:
+            return
```
Comment (datapythonista) [inline feedparser/namespaces/georss.py]: What do you think if besides fixing the bug, we make the implementation a bit more clear and robust?

Something like:
```
def _gen_georss_coords(value, swap=True, dims=2):
    """A generator of (lon, lat) pairs from a string of encoded GeoRSS
    coordinates. Converts to floats and swaps order.
    
    Examples
    --------
    >>> list(_gen_georss_coords('1,10,2,20'))
    [(10.0, 1.0), (20.0, 2.0)]
    """
    coords_list = map(float, value.split(','))
    while True:
        coords = list(itertools.islice(coords_list, dims))
        if not len(coords):
            break
        elif len(coords) == dims:
            if swap:
                coords[:2] = reversed(coords[:2])
            yield tuple(coords)
        else:
            raise ValueError(
                'Inconsistent number of dimensions found. '
                "Can't divide {} coordinates into {} dimensions".format(
                    value.count(',') + 1, dims))
```

**Comment by rgammans** [comment]
I certainly agree  a clearer implementation would be better - i discover this at a spring last night and there were a few boggled eyes.

I'd prefer to change it more like this  - I think the for loop is more readable, and using the grouper pattern from the itertools documentation. Even better might be to explicitly call out the grouper as sub function.

`def _gen_georss_coords(value, swap=True, dims=2):
    """A generator of (lon, lat) pairs from a string of encoded GeoRSS
    coordinates. Converts to floats and swaps order.
    
    Examples
    --------
    >>> list(_gen_georss_coords('1,10,2,20'))
    [(10.0, 1.0), (20.0, 2.0)]
    """
    # Group by Dimension size
    coords_iter = [ map(float, value.split(','))] * dims
    for coords in itertools.izip_longest(*coords_iter):
         if coords[-1] is not None:
            if swap:
                coords[:2] = reversed(coords[:2])
            yield tuple(coords)
         else:
             raise ValueError(
                'Inconsistent number of dimensions found. '
                "Can't divide {} coordinates into {} dimensions".format(
                    value.count(',') + 1, dims))
`

**Comment by datapythonista** [comment]
Yep, my point was to make it a bit clearer and robust, any good implementation will be all right.

Your code looks good to me, but I think your code needs couple of changes to generate the right results. And also it becomes quite more complicated if made compatible with Python 2 and 3. But happy with any approach.

**Comment by rgammans** [comment]
Yup. I missed a iter() out from around the map. Also izip_longest returns tuples. so they need to be converted back to lists for the slice assignment on swap.

**Comment by datapythonista** [comment]
Cool, looking forward to see the PR updated and merged. We already got python.org in Django 2.0, looking forward to see it in Python 3.7. :)

**Comment by balloob** [comment]
What's the status of this PR, what needs to be done to get this through?

**Comment by datapythonista** [comment]
@kurtmckee is this repo still maintained?

**Comment by kurtmckee** [comment]
Hey guys, thanks for working on this. Is there a test case that will demonstrate an unhandled exception prior to merging this? That is, even though PEP 479 changes how StopIteration is managed inside of generators, I'd like to see feedparser crash in the unit tests prior to merging.

Thanks!

**Comment by datapythonista** [comment]
Did you run the unit tests in Python 3.7? If I'm not wrong, the issue was detected with the current unit tests.

**Comment by balloob** [comment]
@kurtmckee the current tests fail under Python 3.7 and work when this patch is applied. What other test case are you thinking of?

**Comment by exxamalte** [comment]
@kurtmckee here's what I've done to showcase how the build fails on Python 3.7 and how this PR fixes the issue:
1. Forked your repository, and added py37 to `tox.ini` and added `.travis.yml` including Python 3.6 and 3.7 tox jobs for comparison.
The Python 3.7 tox job clearly fails: https://travis-ci.org/exxamalte/feedparser/jobs/439063565
2. Then I cherry-picked the commit of this PR. I haven't made any other code changes, nor changes to the unit tests.
The Python 3.7 tox job now also runs successfully: https://travis-ci.org/exxamalte/feedparser/jobs/439086982

The Python 3.6 tox jobs runs successfully with and without this fix.

**Comment by exxamalte** [comment]
> @exxamalte could you please create a PR for travis support - it would be a great help IMO

Done: https://github.com/kurtmckee/feedparser/pull/150

**Comment by answerquest** [comment]
Hi, can the documentation please warn folks not to run this on py3.7 till you do this PR?

**Comment by klevstul** [comment]
@datapythonista @kurtmckee 
any update on the pr?

**Comment by datapythonista** [comment]
I think this repo is unmaintained. I think someone who is using feedparser should create a fork, merge this PR there (and any other ready to be merged in this repo), and projects should start using the fork.

I don't use feedparser myself, this PR was created in a sprrint to migrate www.python.org to Python 3.7, and it has been blocked until now. So, I'm not really the person to create the fork and maintain it.

**Comment by klevstul** [comment]
@datapythonista 
yes, you are right, this repo, unfortunately, seems abandoned. 

maybe one of the [contributors](https://github.com/kurtmckee/feedparser/graphs/contributors) feel like taking on the task of forking and maintaining an updated version? what do you say @twm, @sgillies, @sebest and @paulswartz?

**Comment by kurtmckee** [comment]
Thanks for this work!

**Comment by rpitera** [comment]
Hoping for a release soon, this particular error is keeping me from using this great program. :(

**Comment by LuisValero2000** [comment]
Hey, what will happen if I have the same error with a 3.8.5 version of Python?

**Comment by kurtmckee** [comment]
Nothing will happen without more details. =)

**Pull request merged**